### PR TITLE
RST Reader: parse list table directive

### PR DESCRIPTION
**Pull request by k16shikano**

### CODE REVIEW

File: src/Text/Pandoc/Readers/RST.hs [hunk 1]
```diff
@@ -770,6 +771,26 @@ tableDirective top _fields body = do
                                   aligns' widths' header' rows'
        _ -> return mempty
 
+listTableDirective :: PandocMonad m => String -> [(String, String)] -> String -> RSTParser m Blocks
+listTableDirective top _fields body = do
+  bs <- parseFromString parseBlocks body
+  title <- do { parseFromString (trimInlines . mconcat <$> many inline) top }
+  let rows = takeRows $ B.toList bs
+      numOfCols = length $ rows !! 0
```
Comment (jgm) [inline src/Text/Pandoc/Readers/RST.hs]: Note, `!!` is a partial function. This will fail with a runtime exception if `rows` is empty.  This case should be handled safely.

File: src/Text/Pandoc/Readers/RST.hs [hunk 2]
```diff
@@ -770,6 +771,26 @@ tableDirective top _fields body = do
...
+      headerRows = case lookup "header-rows" _fields of
+                     Just x -> read x :: Int
```
Comment (jgm) [inline src/Text/Pandoc/Readers/RST.hs]: `read` is also partial and will crash at runtime if `x` can't be read as an `Int`. We don't want that.  Check out `safeRead` from `Text.Pandoc.Shared`.

File: src/Text/Pandoc/Readers/RST.hs [hunk 3]
```diff
@@ -770,6 +771,26 @@ tableDirective top _fields body = do
...
+                     Nothing -> 0
+  return $ B.singleton $ Table (B.toList title)
```
Comment (jgm) [inline src/Text/Pandoc/Readers/RST.hs]: Better to use B.table; this will also avoid the need for the `B.toList`s.

File: src/Text/Pandoc/Readers/RST.hs [hunk 4]
```diff
@@ -770,6 +771,26 @@ tableDirective top _fields body = do
...
+                           (replicate numOfCols AlignDefault)
+                           (replicate numOfCols 0)
+                           (if headerRows > 0 then rows !! 0 else replicate numOfCols [])
```
Comment (jgm) [inline src/Text/Pandoc/Readers/RST.hs]: Again `!!` is partial.

**Comment by jgm** [comment]
Thanks for the PR!  I've made some comments that should be addressed.

Also, it would be good to add some tests. The easiest way to do this would be to add `test/command/3432.md` (see the other files in that directory for guidance).

You are aware of #3432, right?  Your commit message should indicate that this commit closes that issue.

We should also support the attributes that list tables can take, such as `widths` and `header-rows`; see #3432 for an example, and see the RST docs for a full list.

**Comment by k16shikano** [comment]
Sorry for the mess and thanks for your comments. I'll fix them later.

File: src/Text/Pandoc/Readers/RST.hs (hunk 4 above)
Comment (veloduff) [inline src/Text/Pandoc/Readers/RST.hs]: I'm not sure if this is the right place for this question, but do you know when this commit will be merged?  I'm right in the middle of producing critical docs, and the lack of list-table for rst is proving to be a big hindrance.

File: src/Text/Pandoc/Readers/RST.hs [hunk 5]
```diff
@@ -775,17 +775,18 @@ tableDirective top _fields body = do
...
 listTableDirective top _fields body = do
   bs <- parseFromString parseBlocks body
-  title <- do { parseFromString (trimInlines . mconcat <$> many inline) top }
+  title <- parseFromString (trimInlines . mconcat <$> many inline) top
   let rows = takeRows $ B.toList bs
-      numOfCols = length $ rows !! 0
-      headerRows = case lookup "header-rows" _fields of
-                     Just x -> read x :: Int
-                     Nothing -> 0
+      headerRow = case rows of
```
Comment (jgm) [inline src/Text/Pandoc/Readers/RST.hs]: I'd suggest something like
```
(firstRow, restRows) = case rows of
                 (x:xs) -> (x, xs)
                 [] -> ([], [])
```
Then you don't have to do the `drop 1 rows` below.  (This should be a *tiny* bit more efficient.)

File: test/command/3432.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+<td>If we took the bones out, it wouldn't be crunchy, now would it?</td>
+</tr>
+<tr class="odd">
+<td>Gannet Ripple</td>
+<td>1.99</td>
+<td>On a stick!</td>
+</tr>
+</tbody>
+</table>
+```
```
Comment (jgm) [inline test/command/3432.md]: This looks great.  I'd just suggest adding a few more tests to cover things like:  no header row, header rows > 1, no widths, empty body or inappropriate body (paragraph rather than list).

Comment (jgm) [inline test/command/3432.md]: PS. You can add further tests in the same file.

Comment (k16shikano) [inline test/command/3432.md]: Thanks for the suggestion. I added tests for below cases. 

* widths is "auto"
* without header-rows
* without widths and with header-rows bigger than 1 (expecting that only the first row is treated as a header) 
* with an empty cell (= cell with just a white space)
* with a cell having a bulletlist in it

File: src/Text/Pandoc/Readers/RST.hs (hunk 5 above)
Comment (k16shikano) [inline src/Text/Pandoc/Readers/RST.hs]: It actually makes the builder part far more concise! Thanks!

File: test/command/3432.md (hunk 6 above)
Comment (jgm) [inline test/command/3432.md]: Great!  I hope you'll contribute more in the future.
